packages/commerce-sdk-react/src/hooks/Test/query.ts

```diff
@@ -0,0 +1,234 @@
+): UseQueryResult<DataType<Client['getBasket']>> => {
+    const {shopperBaskets: client} = useCommerceApi()
+    const method = (arg: Argument<Client['getBasket']>) => client.getBasket(arg)
+    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    // Parameters can be set in `apiOptions` or `client.clientConfig`; they are merged in the helper
+    // hook, so we use a callback here that receives that merged object.
+    const getQueryKey = <T extends Record<string, unknown>>(parameters: T) =>
+        [
+            'https://',
+            parameters.shortCode,
+            '.api.commercecloud.salesforce.com/checkout/shopper-baskets/',
+            parameters.version,
+            '/organizations/',
+            parameters.organizationId,
+            '/baskets/',
+            parameters.basketId,
+            '?',
+            'siteId',
+            parameters.siteId,
+            'locale',
+            parameters.locale,
+            // Full parameters last for easy lookup
+            parameters
+        ] as const
+
+    return useQuery(apiOptions, queryOptions, {
+        client,
+        method,
+        requiredParameters,
+        getQueryKey
+    })
+}
+/**
+ * A hook for `ShopperBaskets#getPaymentMethodsForBasket`.
+ * Gets applicable payment methods for an existing basket considering the open payment amount only.
+ * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=getPaymentMethodsForBasket} for more information about the API endpoint.
+ * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#getpaymentmethodsforbasket} for more information on the parameters and returned data type.
+ * @returns An object describing the state of the request.
+ */
+export const usePaymentMethodsForBasket = (
+    apiOptions: Argument<Client['getPaymentMethodsForBasket']>,
+    queryOptions: Omit<
+        UseQueryOptions<DataType<Client['getPaymentMethodsForBasket']>>,
+        'queryFn'
+    > = {}
+): UseQueryResult<DataType<Client['getPaymentMethodsForBasket']>> => {
+    const {shopperBaskets: client} = useCommerceApi()
+    const method = (arg: Argument<Client['getPaymentMethodsForBasket']>) =>
+        client.getPaymentMethodsForBasket(arg)
+    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    // Parameters can be set in `apiOptions` or `client.clientConfig`; they are merged in the helper
+    // hook, so we use a callback here that receives that merged object.
+    const getQueryKey = <T extends Record<string, unknown>>(parameters: T) =>
+        [
+            'https://',
+            parameters.shortCode,
+            '.api.commercecloud.salesforce.com/checkout/shopper-baskets/',
+            parameters.version,
+            '/organizations/',
+            parameters.organizationId,
```

Comment:
I assume here that we set those values as queryKey is because we want to cache different sets of commerce config. Would this be a bit overkill?

Reply:
It definitely is overkill; I wasn't sure what data we wanted to capture for query keys, so I started off by capturing it all. After a bit more exposure to the code, I'm thinking a good idea would be to have the filled-out template path and an object at the end with path + query parameters. (`['/baskets/', parameters.basketId, /* ..., */ parameters]`) I think that this will give us the flexibility we need (e.g. invalidate most `/baskets/{basketId}`, but selectively update those with matching `locale` query parameter so we don't get the wrong language somewhere) while not having too cumbersome a syntax (e.g. having to remember the order of the query parameters, as exists in the current state of this PR).